recipes/octo-logger-cpp/all/conanfile.py

```diff
@@ -28,6 +34,14 @@ def _compilers_minimum_version(self):
             "msvc": "1923",
         }
 
+    @property
+    def _aws_supported(self):
+        return Version(self.version) >= "1.2.0"
+
+    def configure(self):
+        if self.options.with_aws and self._aws_supported:
+            self.options["aws-sdk-cpp"].logs = True
```

Comment:
```suggestion
        if not self._aws_supported:
            del self.options.with_aws
```

Older versions should not have that option.
Move the mandatory dependency option to validate() method:


```
def validate(self):
    if self.info.options.get_safe("with_aws"):
        if not self.dependencies["aws-sdk-cpp"].options.logs:
            raise ConanInvalidConfiguration(f"{self.name} requires the option aws-sdk-cpp:logs=True")
```

Reply:
gotcha, fixed both